core/src/test/java/org/apache/iceberg/rest/TestRESTViewCatalog.java

```diff
@@ -144,6 +157,65 @@ public void closeCatalog() throws Exception {
     }
   }
 
+  @Test
+  public void testPaginationForListViews() {
+    RESTCatalogAdapter adapter = Mockito.spy(new RESTCatalogAdapter(backendCatalog));
+    RESTCatalog catalog =
+        new RESTCatalog(SessionCatalog.SessionContext.createEmpty(), (config) -> adapter);
+    catalog.initialize("test", ImmutableMap.of(RESTSessionCatalog.REST_PAGE_SIZE, "10"));
+
+    int numberOfItems = 20;
+    int numberOfInvocations = 2;
+    String namespaceName = "newdb";
+    String viewName = "newview";
+
+    // create initial namespace
+    catalog().createNamespace(Namespace.of(namespaceName));
+
+    // create several views under namespace, based off a table for listing and verify
+    for (int i = 0; i < numberOfItems; i++) {
+      TableIdentifier viewIndentifier = TableIdentifier.of(namespaceName, viewName + i);
+      catalog
+          .buildView(viewIndentifier)
+          .withSchema(SCHEMA)
+          .withDefaultNamespace(viewIndentifier.namespace())
+          .withQuery("spark", "select * from ns.tbl")
+          .create();
+    }
+    List<TableIdentifier> views = catalog.listViews(Namespace.of(namespaceName));
+    assertThat(views).hasSize(numberOfItems);
+
+    Mockito.verify(adapter)
+        .execute(
+            eq(HTTPMethod.GET),
+            eq("v1/config"),
+            any(),
+            any(),
+            eq(ConfigResponse.class),
+            any(),
+            any());
+
+    Mockito.verify(adapter, times(numberOfItems))
+        .execute(
+            eq(HTTPMethod.POST),
+            eq(String.format("v1/namespaces/%s/views", namespaceName)),
+            any(),
+            any(),
+            eq(LoadViewResponse.class),
+            any(),
+            any());
+
+    Mockito.verify(adapter, times(numberOfInvocations))
+        .execute(
+            eq(HTTPMethod.GET),
+            eq(String.format("v1/namespaces/%s/views", namespaceName)),
+            eq(ImmutableMap.of("pageToken", "10", "pageSize", "10")),
```

Comment:
rather than verifying that there have been `numberOfInvocations` where `pageToken` / `pageSize` are 10, I think there should be a separate check where `pageToken = ""`/ `pageSize = "10"`

Reply:
Yea I had tried out a similar example to what you shared above, and mentioned the issues i faced here https://github.com/apache/iceberg/pull/9782#discussion_r1571188458, when i tried make two seperate requests, like so :

```

    Mockito.verify(adapter)
        .execute(
            eq(HTTPMethod.GET),
            eq("v1/namespaces"),
            eq(ImmutableMap.of("pageToken", "", "pageSize", "10")),
            any(),
            eq(ListNamespacesResponse.class),
            any(),
            any());

    Mockito.verify(adapter)
            .execute(
                    eq(HTTPMethod.GET),
                    eq("v1/namespaces"),
                    eq(ImmutableMap.of("pageToken", "10", "pageSize", "10")),
                    any(),
                    eq(ListNamespacesResponse.class),
                    any(),
                    any());

``` 

It  always fail on the first request, i think because this list namespaces endpoint is hit multiple times and the last value of the query params is `"pageToken", "10", "pageSize", "10"` is what gets called thus leading to this exception, instead of what was sent intially.

```
Argument(s) are different! Wanted:
rESTCatalogAdapter.execute(
    GET,
    "v1/namespaces",
    {"pageToken" = "", "pageSize" = "10"},
    <any>,
    class org.apache.iceberg.rest.responses.ListNamespacesResponse,
    <any>,
    <any>
);
```

Reply:
@nastra I am able to make two separate requests with `.handleRequest` instead of `execute`, and send the correct query params for each call.